lib/driver.js

```diff
@@ -281,10 +292,35 @@ class AndroidUiautomator2Driver extends BaseDriver {
     if (!this.opts.systemPort) {
       return;
     }
-    await PORT_ALLOCATION_GUARD.acquire(AndroidUiautomator2Driver.name,
+    await DEVICE_PORT_ALLOCATION_GUARD.acquire(AndroidUiautomator2Driver.name,
       async () => await this.adb.removePortForward(this.opts.systemPort));
   }
 
+  async allocateMjpegServerPort () {
+    await MJPEG_SERVER_PORT_ALLOCATION_GUARD.acquire(AndroidUiautomator2Driver.name, async () => {
+      try {
+        this.opts.mjpegServerPort = this.opts.mjpegServerPort || await findAPortNotInUse(
+          MJPEG_SERVER_PORT_RANGE[0], MJPEG_SERVER_PORT_RANGE[1]);
+      } catch (e) {
+        logger.errorAndThrow(
```

Comment:
I'd rather print a warning und unset this.opts.mjpegServerPort capability to make it less intrusive

Reply:
This is resolved with the latest commit.